vector/src/main/java/im/vector/app/core/pushers/model/PushDataFcm.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.pushers.model
+
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonClass
+
+/**
+ * In this case, the format is:
+ * <pre>
+ * {
+ *     "event_id":"$anEventId",
+ *     "room_id":"!aRoomId",
+ *     "unread":"1",
+ *     "prio":"high",
+ * }
+ * </pre>
+ * .
```

Comment:
@p1gp1g do you know why the Json format is not the same between FCM and other distibutors? (See `PushDataUnifiedPush`)

Reply:
Notifications received by FCM are formatted by the matrix gateway [1]. The data send to FCM is the content of the "notification" attribute of the json sent to the gateway [2][3].
On the other side, with UnifiedPush, the content of the message received is the content posted to the push gateway endpoint [3].

If we want to get the same content with FCM and unifiedpush, we can do a new sygnal pusher [4].

[1] https://github.com/matrix-org/sygnal/blob/main/sygnal/gcmpushkin.py
[2] https://github.com/matrix-org/sygnal/blob/main/sygnal/gcmpushkin.py#L366
[3] https://spec.matrix.org/latest/push-gateway-api/
[4] https://github.com/p1gp1g/sygnal/blob/unifiedpush/sygnal/upfcmpushkin.py (Not tested for a while)

Reply:
Thanks. Copied to the code: 9c5c43fefcdeb2d002954a3feee2fadee58a9c81